collada/__init__.py

```diff
@@ -286,8 +286,20 @@ def _loadGeometry(self):
                     for geomnode in libnode.findall(tag('geometry')):
                         if geomnode.find(tag('mesh')) is None:
                             continue
+
+                        name = ''
+                        lib_nodenames = self.xmlnode.find(tag('library_nodes'))
+                        if lib_nodenames is not None:
+                            for nodename in lib_nodenames.findall(tag('node')):
+                                instance = nodename.find(tag('instance_geometry'))
```

Comment:
I don't think you're making this change in the right place here. I think you want to make the change in scene.py where the Node is loaded.